scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScaladocParser.scala

```diff
@@ -235,6 +236,13 @@ object ScaladocParser {
       tableParser |
       leadTextParser // keep at the end, this is the fallback
 
+  private def embeddedTermsParser[_: P](indent: Int): P[Seq[Term]] = P {
+    (listBlockParser(indent) |
+      codeBlockParser |
```

Comment:
`codeBlockParser` should take indent too.

I haven't covered all cases in tests but the additional ones are:
~~~
// case 1 - no indent
- item  
```
code
```
// result
Paragraph(
   List(
      ListBlock(...),
      MdCodeBlock(..)
   )
)
------
// case 2 - larger than list indent + 3
- item  
      ```
      code
      ```
// result
Paragraph(
   List(
      ListBlock(
        ListItem(List(Text(Word("item"), Word("```"), ...)))
      )
   )
)
~~~

Reply:
> `codeBlockParser` should take indent too.

what makes you think that? and what should be indented, the delimiters `{{{` or the entire block?

what about table? should that be indented, too (and according to what spec)?

Reply:
Oh, I confused it with mdcodeblock. I think in this case it would be easier to review to whole changeset to list item parsing than splitting it onto several PRs.

What about tables - probably they should be indented. 
There is no spec on it but I think it should work in a similar way as fenced codeblocks/quotes in markdown (like this one:  [example 254](https://spec.commonmark.org/0.30/#example-254))

Reply:
my read is that they should not.

if scala3 implemented the full markdown, we have to write a completely new parser; this one doesn't support blockquotes, md lists (which use different different markers and increment), headings with other markers etc.

so, I'm considering this just an extension of scala 2 with fenced code blocks. and in scala 2, they are not documented to be indented, and i don't know what the actual scaladoc tool does (maybe that is what we should find out, as documentation is not very comprehensive).

Reply:
and I will update this pr with additional commits (but do read them one at a time, should be easier).

however, i was hoping you'd review the other pre first, so i can rebase and adjust.

Reply:
rebased, added the other two commits